packages/tpetra/CMakeLists.txt

```diff
@@ -77,25 +77,30 @@ IF (Tpetra_ENABLE_CUDA)
-  IF (DEFINED Trilinos_CXX_FLAGS AND Trilinos_CXX_FLAGS MATCHES "-expt-extended-lambda")
-    # mfh 30 Sep 2017: Even if you put the flag in Trilinos_CXX_FLAGS,
-    # Trilinos seems to move that flag into CMAKE_CXX_FLAGS for some
-    # reason.  Nevertheless, it doesn't hurt to check both.
-    SET(TpetraCore_FOUND_CUDA_LAMBDA_FLAG ON)
-  ELSEIF (DEFINED CMAKE_CXX_FLAGS AND CMAKE_CXX_FLAGS MATCHES "-expt-extended-lambda")
-    SET(TpetraCore_FOUND_CUDA_LAMBDA_FLAG ON)
-  ENDIF ()
-  IF (NOT TpetraCore_FOUND_CUDA_LAMBDA_FLAG)
-    MESSAGE (FATAL_ERROR "If building with CUDA, Tpetra and downstream packages require, at least with CUDA_VERSION >= 7.5 and <= 9.0, that you add \"--expt-extended-lambda\" to either Trilinos_CXX11_FLAGS or CMAKE_CXX_FLAGS.  You must also set Kokkos_ENABLE_Cuda_Lambda:BOOL=ON, if it is not already ON by default.")
-  ENDIF ()
-
-  IF (NOT DEFINED Kokkos_ENABLE_Cuda_Lambda OR NOT Kokkos_ENABLE_Cuda_Lambda)
-    MESSAGE (FATAL_ERROR "If building with CUDA, Tpetra and downstream packages require that you set the CMake option Kokkos_ENABLE_Cuda_Lambda:BOOL=ON, if it is not already ON by default.  For details, please refer to Trilinos issue #1682 (https://github.com/trilinos/Trilinos/issues/1682).  You must also add \"--expt-extended-lambda\" to either Trilinos_CXX11_FLAGS or CMAKE_CXX_FLAGS, at least with CUDA_VERSION >= 7.5 and <= 9.0.")
+  # jjw 25 Oct 2019: It is no longer necessary to verify flags in this way
+  # As long as Kokkos_ENABLE_CUDA_LAMBDA is set, then Kokkos will transitively add all
+  # flags necessary for correctness via target_link_libraries
+  #SET (TpetraCore_FOUND_CUDA_LAMBDA_FLAG OFF)
+  #IF (DEFINED Trilinos_CXX_FLAGS AND Trilinos_CXX_FLAGS MATCHES "-expt-extended-lambda")
+  #  # mfh 30 Sep 2017: Even if you put the flag in Trilinos_CXX_FLAGS,
+  #  # Trilinos seems to move that flag into CMAKE_CXX_FLAGS for some
+  #  # reason.  Nevertheless, it doesn't hurt to check both.
+  #  SET(TpetraCore_FOUND_CUDA_LAMBDA_FLAG ON)
+  #ELSEIF (DEFINED CMAKE_CXX_FLAGS AND CMAKE_CXX_FLAGS MATCHES "-expt-extended-lambda")
+  #  SET(TpetraCore_FOUND_CUDA_LAMBDA_FLAG ON)
+  #ENDIF ()
+  #IF (NOT TpetraCore_FOUND_CUDA_LAMBDA_FLAG)
+  #  MESSAGE (FATAL_ERROR "If building with CUDA, Tpetra and downstream packages require, at least with CUDA_VERSION >= 7.5 and <= 9.0, that you add \"--expt-extended-lambda\" to either Trilinos_CXX11_FLAGS or CMAKE_CXX_FLAGS.  You must also set Kokkos_ENABLE_CUDA_LAMBDA:BOOL=ON, if it is not already ON by default.")
+  #ENDIF ()
+
+  IF (NOT DEFINED Kokkos_ENABLE_CUDA_LAMBDA OR NOT Kokkos_ENABLE_CUDA_LAMBDA)
+    MESSAGE (FATAL_ERROR "If building with CUDA, Tpetra and downstream packages require that you set the CMake option Kokkos_ENABLE_CUDA_LAMBDA:BOOL=ON, if it is not already ON by default.  For details, please refer to Trilinos issue #1682 (https://github.com/trilinos/Trilinos/issues/1682).  You must also use at least with CUDA_VERSION >= 7.5 and <= 9.0.")
```

Comment:
The last sentence should read: "You must also have at least `CUDA_VERSION >= 7.5`.  Thanks!

Reply:
@mhoemmen resolved by merge of VOTD develop back into this PR

Reply:
Oop, missed this one, sorry @mhoemmen ! Will push in a fix to the kokkos-promotion branch

Reply:
Fixed now.